compiler/src/iree/compiler/Codegen/Common/TransformExtensions/CommonExtensions.cpp

```diff
@@ -1024,5 +1026,229 @@ transform_dialect::IREEEraseHALDescriptorTypeFromMemRefOp::apply(
   return DiagnosedSilenceableFailure::success();
 }
 
+//===---------------------------------------------------------------------===//
+// RegisterMatchCallbacksOp
+//===---------------------------------------------------------------------===//
+
+/// Match callback for "_test_match_callback" hook. Matches any payload
+/// operations associated with operand handles unless they have the
+/// "test.iree_transform_do_not_match" attribute, in which case produces a
+/// silenceable failure.
+static DiagnosedSilenceableFailure testMatchCallbackCallback(
```

Comment:
I am assuming everything here is to be moved upstream? If so can we add these to `llvm-external-projects`?

Reply:
Thomas proposed to move stuff as a followup to help distribute the load.
Leaving that to him, thanks Thomas!